app/app/Http/Controllers/ChangePasswordController.php

```diff
@@ -32,36 +33,26 @@ public function post(Request $request)
         if (BaseUser::isSignedIn())
         {
 			$user = BaseUser::getDbUser();
-			$validation_rules = array(
-				'current_password'     => 'required',
-				'new_password'         => 'required',
-				'password_confirm'     => 'required|same:new_password'
-			);
-			$validator = Validator::make(Input::all(), $validation_rules);
-			$failing = $validator->fails();
-			if (!$failing && !empty($user->password_hash))
-			{
-				if ( !BaseUser::authenticate($user->email, $request->input('current_password')) )
-				{
-					$validator->errors()->add('current_password', 'Current password not matched');
-					$failing = true;
-				}
+			$validator = Validator::make(
+			    $request->all(), [
+			        'current_password' => 'required',
+                    'new_password' => 'required',
+                    'password_confirm' => 'required|same:new_password'
+                ]
+            );
+			if ($validator->fails()) {
+			    return redirect()->to('change-password')->withErrors($validator->errors())->withInput();
+            } else {
+			    if (Hash::check($request->get('current_password'), $user->password_hash)) {
```

Comment:
BaseUser::authenticate does the Hash::check in a central place so why did you replace with Hash class directly here?

Reply:
Because I just wanted to check the password don't wanted user to login.

Reply:
BaseUser::authenticate doesn't log a user in.  It just returns true or false indicating if the email and password matched a user.  authenticate doesn't have any side effects.

signin is the method you're talking about which saves that the user signed in.